apps/server/src/shared/infra/identity-management/keycloak-identity-management.ts

```diff
@@ -0,0 +1,128 @@
+import { IAccount, IAccountUpdate } from '@shared/domain';
+
+/* eslint-disable no-nested-ternary */
+import { Inject, Injectable } from '@nestjs/common';
+import KcAdminClient from '@keycloak/keycloak-admin-client';
+import UserRepresentation from '@keycloak/keycloak-admin-client/lib/defs/userRepresentation';
+import { GrantTypes } from '@keycloak/keycloak-admin-client/lib/utils/auth';
+import { IIdentityManagement } from './identity-management.interface';
+
+interface KcSettings {
+	realmName: string;
+	baseUrl: string;
+	credentials: KcCredentials;
+	sourceRealm: string;
+}
+
+interface KcCredentials {
+	grantType: GrantTypes;
+	username: string;
+	password: string;
+	clientId: string;
+}
+
+@Injectable()
+export class KeycloakIdentityManagement implements IIdentityManagement {
+	private lastAuthorizationTime = 0;
+
+	private static AUTHORIZATION_TIMEBOX_MS = 59 * 1000;
+
+	public constructor(
+		@Inject('KeycloakAdminClient') private readonly kcAdminClient: KcAdminClient,
+		@Inject('KeycloakSettings') private readonly kcSettings: KcSettings
+	) {
+		kcAdminClient.setConfig({
```

Comment:
The configuration of the keycloak client should not be done by the class requiring it. Multiple classes might have the same dependency. It should be done in the module providing the instance. 
Using something like this might be an option: https://docs.nestjs.com/fundamentals/custom-providers#factory-providers-usefactory

Reply:
As discussed, we consider the module closed. Now, the abstract IdentityManagement is exported.